commcare_connect/opportunity/visit_import.py

```diff
@@ -311,3 +313,173 @@ def get_status_by_completed_work_id(dataset):
+        self.area_name = self._get_area_name()
+        self.username = self._get_username()
+        self.id = self._get_id()
+
+    def _get_latitude(self) -> Decimal:
+        error_message = "Latitude must be between -90 and 90 degrees."
+        index = _get_header_index(self.headers, LATITUDE_COL)
+        try:
+            latitude = Decimal(self.row[index])
+        except (ValueError, TypeError, InvalidOperation):
+            raise InvalidValueError(error_message)
+        if not Decimal("-90") <= latitude <= Decimal("90"):
+            raise InvalidValueError(error_message)
+        return latitude
+
+    def _get_longitude(self) -> Decimal:
+        index = _get_header_index(self.headers, LONGITUDE_COL)
+        try:
+            longitude = Decimal(self.row[index])
+        except (ValueError, TypeError, InvalidOperation):
+            raise InvalidValueError(f"Invalid longitude value: {self.row[index]}")
+        if not Decimal("-180") <= longitude <= Decimal("180"):
+            raise InvalidValueError("Longitude must be between -180 and 180 degrees")
+        return longitude
+
+    def _get_radius(self) -> int:
+        index = _get_header_index(self.headers, RADIUS_COL)
+        try:
+            radius = int(self.row[index])
+        except (ValueError, TypeError):
+            raise InvalidValueError(f"Invalid radius value: {self.row[index]}")
+        if radius <= 0:
+            raise InvalidValueError("Radius must be a positive integer")
+        return radius
+
+    def _get_active(self) -> bool:
+        error_message = "Active status must be 'yes' or 'no'"
+        index = _get_header_index(self.headers, ACTIVE_COL)
+        active = self.row[index]
+        if not active:
+            raise InvalidValueError(error_message)
+        active = active.lower().strip()
+        if active not in ["yes", "no"]:
+            raise InvalidValueError(error_message)
+        return active == "yes"
+
+    def _get_area_name(self) -> str:
+        index = _get_header_index(self.headers, AREA_NAME_COL)
+        area_name = self.row[index]
+        if not area_name:
+            raise InvalidValueError("Area name is not valid.")
+        return area_name
+
+    def _get_username(self) -> str | None:
+        try:
+            index = _get_header_index(self.headers, USERNAME_COL)
+        except ImportException:
+            return None
+        username = self.row[index]
+        return username if username else None
+
+    def _get_id(self) -> int | None:
+        try:
+            index = _get_header_index(self.headers, ID)
+        except ImportException:
+            return None
+        catchment_id = self.row[index]
+        return catchment_id if catchment_id else None
+
+
+def create_or_update_catchment(row_data: RowData, opportunity: Opportunity, opportunity_access: dict):
+    if row_data.id:
+        catchment = None
+        try:
+            catchment = CatchmentArea.objects.get(id=row_data.id)
```

Comment:
It would be great if this enforced that the model was in the right opportunity, especially since the id is a simple int, and its easy to guess other people's ids.

Reply:
On that note, I am a bit uncomfortable matching on id like this, since it is difficult to verify and easy to mess up. How would you feel about matching on name, or adding a new "site_code" to the model that would be a human readable unique identifier (like a slug)

Reply:
@calellowitz We can match by name, but no two catchment areas can have the same name within an opportunity, right?